scripts/ui.ts

```diff
@@ -111,47 +113,69 @@ async function requestLoginAccess(permissions: Permission[]): Promise<[boolean,
     throw new Error(reason);
   }
 
+  // Get the seed and email.
+
   let seed = checkStoredSeed();
 
+  // If we don't have a seed, get it from the seed provider. We will also get
+  // the email if the user is signing up for the first time.
   if (!seed) {
-    // If we don't have a seed, show seed provider chooser.
+    // Show seed provider chooser.
 
     const seedProviderDisplayUrl = await getSeedProviderDisplayUrl();
 
     // User has chosen seed provider, open seed provider display.
 
     log("Calling runSeedProviderDisplay");
-    seed = await runSeedProviderDisplay(seedProviderDisplayUrl);
+    let email = null;
+    [seed, email] = await runSeedProviderDisplay(seedProviderDisplayUrl);
+
+    // Register and get the JWT.
+    let jwt = null;
+    if (email) {
+      try {
+        jwt = await register(client, seed, email);
+      } catch (e1) {
+        try {
+          jwt = await login(client, seed, email);
```

Comment:
F/U: Make these two calls in parallel, as we discussed. It will speed things up a lot.